### PR TITLE
fix(pci.projects.storage): check if backup is installed before deletion

### PR DESCRIPTION
**Pull request by frenautvh**
See : MSB-133

issue : when customer is trying to delete an instance snapshot, if the snapshot is installed the api call fails so the delete button should be disabled

fix: display a warning message and disable the delete button if the backup is installed on an instance

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/storages/instance-backups/instance-backup/delete/delete.controller.js [hunk 1]
```diff
@@ -11,7 +11,18 @@ export default class PciBlockStorageDetailsDeleteController {
...
 
   $onInit() {
-    this.isLoading = false;
+    this.isLoading = true;
+    this.isBackupDeletable = this.instanceBackup.isDeletable();
+    this.PciProjectStorageInstanceBackupService
+      .isInstalled(this.projectId, this.instanceBackup)
+      .then((installed) => {
+        if (installed) {
+          this.isBackupDeletable = false;
```
Comment (FredericEspiau) [inline packages/manager/modules/pci/src/projects/project/storages/instance-backups/instance-backup/delete/delete.controller.js]:
```suggestion
          this.isBackupDeletable = !installed;
```
?

File: packages/manager/modules/pci/src/projects/project/storages/instance-backups/instance-backup/delete/delete.controller.js [hunk 2]
```diff
@@ -11,7 +11,18 @@ export default class PciBlockStorageDetailsDeleteController {
   }
 
   $onInit() {
-    this.isLoading = false;
+    this.isLoading = true;
+    this.isBackupDeletable = this.instanceBackup.isDeletable();
+    this.PciProjectStorageInstanceBackupService
```
Comment (FredericEspiau) [inline packages/manager/modules/pci/src/projects/project/storages/instance-backups/instance-backup/delete/delete.controller.js]:
```suggestion
    return this.PciProjectStorageInstanceBackupService
```

**Comment by frenautvh** [comment]
Update : add to do some rework on this PR to match QC requirements (displaying instance name in error message)

**Comment by ovh-cds** [comment]
CDS Report telecom-task#3147.0 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report test-manager_2#3147.0 ■   
* Stage 1
  * e2e ■   
  * lint ✔

**Comment by ovh-cds** [comment]
CDS Report overthebox#3147.0 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report pci#3147.0 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report test-manager_2#3147.0 ■   
* Stage 1
  * e2e ■   
  * lint ✔

**Comment by ovh-cds** [comment]
CDS Report telecom-dashboard#3147.0 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report sms#3147.0 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report freefax#3147.0 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report layout-ovh#3147.0 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report vrack#3147.0 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report public-cloud#3147.0 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report test-manager_2#3147.0 ■   
* Stage 1
  * e2e ■   
  * lint ✔

**Comment by ovh-cds** [comment]
CDS Report installation#3162.0 ✘  
* Stage 1
  * Install ✘

File: packages/manager/modules/pci/src/projects/project/storages/instance-backups/instance-backup/delete/delete.controller.js [hunk 3]
```diff
@@ -11,7 +11,15 @@ export default class PciBlockStorageDetailsDeleteController {
   }
 
   $onInit() {
-    this.isLoading = false;
+    this.isLoading = true;
+    return this.PciProjectStorageInstanceBackupService
+      .getAssociatedInstance(this.projectId, this.instanceBackup)
+      .then((installed) => {
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/storages/instance-backups/instance-backup/delete/delete.controller.js]:
```suggestion
      .then((instance) => {
```
? Looks more logical that way

File: packages/manager/modules/pci/src/projects/project/storages/instance-backups/instance-backup/delete/delete.html [hunk 4]
```diff
@@ -20,4 +20,11 @@
             <span data-translate="pci_projects_project_storages_volume-instances_volume-instance_delete_error_not_deletable"></span>
         </oui-message>
     </div>
+    <div data-ng-if="$ctrl.associatedInstances.length > 0">
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/storages/instance-backups/instance-backup/delete/delete.html]: You can apply the ngIf directive directly on the `<oui-message>` component I guess.

File: packages/manager/modules/pci/src/projects/project/storages/instance-backups/instance-backup/delete/delete.controller.js [hunk 5]
```diff
@@ -11,7 +12,15 @@ export default class PciBlockStorageDetailsDeleteController {
   }
 
   $onInit() {
-    this.isLoading = false;
+    this.isLoading = true;
+    return this.PciProjectStorageInstanceBackupService
+      .getAssociatedInstances(this.projectId, this.instanceBackup)
+      .then((instances) => {
+        this.associatedInstances = map(instances, 'name');
```
Comment (FredericEspiau) [inline packages/manager/modules/pci/src/projects/project/storages/instance-backups/instance-backup/delete/delete.controller.js]:
```suggestion
        this.associatedInstanceNames = map(instances, 'name');
```
;)

Comment (frenautvh) [reply packages/manager/modules/pci/src/projects/project/storages/instance-backups/instance-backup/delete/delete.controller.js]: done ty

File: packages/manager/modules/pci/src/projects/project/storages/instance-backups/instance-backup/delete/delete.html (hunk 4 above)
Comment (frenautvh) [reply packages/manager/modules/pci/src/projects/project/storages/instance-backups/instance-backup/delete/delete.html]: done ty

**Pull request merged**